doc/openssl_migration_guide.rst

```diff
@@ -0,0 +1,389 @@
+OpenSSL 1.1 to Botan 3.x Migration
+==================================
+
+This aims to be a rough guide for migrating applications from OpenSSL 1.1 to Botan 3.x.
+
+This guide attempts to be, but is not, complete. If you run into a problem while
+migrating code that does not seem to be described here, please open an issue on
+`GitHub <https://github.com/randombit/botan/issues>`_.
+
+General Remarks
+----------------
+
+* Botan is a C++ library, whereas OpenSSL is a C library
+* Botan also provides a :doc:`C API <api_ref/ffi>` for most of its functionality,
```

Comment:
Is there a difference between the `:doc:` and `:ref:` markups?

Reply:
With `:doc:` you reference whole documents, with `:ref:` you reference sections. See https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#cross-referencing-using-roles.